lib/sort_by_length.rb

```diff
@@ -1,7 +1,68 @@
 # A method which will return an array of the words in the string
-#  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# sorted by the length of the word.
+#
+# Where n is the size of the array (or number of words passed through the string):
+# >> Time complexity: O(nlogn)
+#    There are logn levels of work to do (dividing array in half logn times)
+#    At each level, a total of n - 1 comparisons are made
+#    (on average - unless first element of right array is larger than the last element in the left array)
+# >> Space complexity: O(n)
+#    We create temporary left and right sub-arrays (by splitting original in half each layer)
+#    However, due to Ruby garbage collection, we only need space for the final merged array of size n
+
+def split_arr(arr)
+  mid_point = (arr.size - 1) / 2
+
+  left_arr = arr[0..mid_point]
+  right_arr = arr[(mid_point + 1)..(arr.size - 1)]
+
+  return left_arr, right_arr
+end
+
+def merge_and_sort(left, right)
+  merged = []
+
+  while left.size > 0 && right.size > 0
+    if left[0].length < right[0].length
+      merged << left[0]
+      left.shift
+    elsif left[0].length == right[0].length
+      merged << left[0]
+      left.shift
+    else
+      merged << right[0]
+      right.shift
```

Comment:
Just note that `.shift` is O(n), so your method ends up being O(n^2)

Reply:
Oh!! Tricky tricky...